### PR TITLE
serial consoles: make it possible to override the default and add mor…

### PR DESCRIPTION
**Pull request by Mic92**
…e documentation for it.

### CODE REVIEW

File: common/serial.nix [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+      example = [ "ttyS2,115200" ];
+      description = lib.mdDoc ''
+        The Linux kernel console option allows you to configure various devices as
+        consoles. The default setting is configured to provide access to serial
+        consoles, such as IPMI SOL console redirection found in BMCs or GPIO-based
+        serial terminals found in embedded devices. You can specify multiple `console=`
+        options on the kernel command line, which will result in output appearing on all
+        of them. The last device specified will be used when opening /dev/console. This
+        information can be found in the Linux documentation at
+        https://www.kernel.org/doc/html/v4.14/admin-guide/serial-console.html.
```
Comment (zimbatm) [inline common/serial.nix:24]:
```suggestion
        <https://www.kernel.org/doc/html/v4.14/admin-guide/serial-console.html>.
```
This depends on the flavor of Markdown but I think a majority prefer the angle quotes.

**Review by zimbatm — APPROVED** [review]
:100:

File: common/serial.nix [hunk 2]
```diff
@@ -1,17 +1,73 @@
...
-
-  # also make grub respond on serial consoles
-  boot.loader.grub.extraConfig = ''
-    serial --unit=0 --speed=115200 --word=8 --parity=no --stop=1
-    terminal_input --append serial
-    terminal_output --append serial
+{ config, lib, pkgs, ... }:
+let
+    # Based on https://unix.stackexchange.com/questions/16578/resizable-serial-console-window
+  resize = pkgs.writeScriptBin "resize" ''
```
Comment (Mic92) [inline common/serial.nix:4]: This makes it much saner...

**Comment by zimbatm** [comment]
bors merge

**Pull request merged**